search-app/src/cmr/search/data/complex_to_simple_converters/spatial.clj

```diff
@@ -214,9 +214,10 @@
           operator (extract-operator (:query-params context))
           spatial-script (shape->script-cond shape operator)
         ; check the MBR first before doing the more expensive check
-          spatial-cond (if (= operator :any)
-                         (gc/and-conds [mbr-cond (gc/or-conds [lr-cond spatial-script])])
-                         spatial-script)]
+          spatial-cond (if (or (= operator :every) (= operator :ignore_br))
```

Comment:
This code looks wrong. The operator should not have anything to do with we match the MBR first before doing the more expensive exactly spatial shape matching.

Reply:
The reason this is needed is because if they have the `:every` or `:ignore_br` options, because for the granules which have the global bounding rectangles in their geometry data, the match MBR first operation will match with the global bounding rectangle doing this initial operation and return the granule even though the `:ignore_br` option is used.

Reply:
It shouldn't. The MBR condition is to filter out the granules that does not have a chance to match with the real spatial condition and cut down the number of times that we have to run the spatial script.

Reply:
This MBR is for the spatial parameter, not the granule's spatial coverage.